Casks/julia.rb

```diff
@@ -1,15 +1,23 @@
 cask "julia" do
-  version "1.6.4"
-  sha256 "a00598f04832b8f5d28039942280feadf40e8a14d7f6264c754ec70627507ace"
+  arch = Hardware::CPU.intel? ? "mac64" : "macaarch64"
+  arch_folder = Hardware::CPU.intel? ? "x64" : "aarch64"
```

Comment:
```suggestion
  arch, arch_folder = Hardware::CPU.intel? ? ["mac64", "x64"] : ["macaarch64", "aarch64"]
```
You can probably even combine this with the `if ... else ... end` block below, though that would be a bit nonstandard.